src/main/java/ua/lviv/iot/lab2/writer/DwellingWriter.java

```diff
@@ -11,21 +13,25 @@
 
 public class DwellingWriter extends DwellingManager {
     public static void writeToFile(List<Dwelling> dwellings) throws IOException {
-        File csvOutputFile = new File("src/main/resources/DwellingsWriter.csv");
-        Charset charSet = StandardCharsets.UTF_8;
-        try (BufferedWriter bufferedWriter = Files.newBufferedWriter(
-                csvOutputFile.toPath(), charSet)) {
-            String previousClassName = "";
-            dwellings.sort(Comparator.comparing(dwelling -> dwelling.getClass().getName()));
-            for (var dwelling : dwellings) {
-                if (!dwelling.getClass().getSimpleName().equals(previousClassName)) {
-                    bufferedWriter.write(dwelling.getHeaders());
+        if (!dwellings.isEmpty()) {
+            File csvOutputFile = new File("src/test/resources/DwellingsWriter.csv");
+            Charset charSet = StandardCharsets.UTF_8;
+            try (BufferedWriter bufferedWriter = Files.newBufferedWriter(
+                    csvOutputFile.toPath(), charSet)) {
+                String previousClassName = "";
+                dwellings.sort(Comparator.comparing(dwelling -> dwelling.getClass().getName()));
+                for (var dwelling : dwellings) {
```

Comment:
Maybe it would be better to use an abstract father of all **_dwellings_** instead of **_var_**